mm2src/coins/z_coin/z_rpc.rs

```diff
@@ -174,17 +160,21 @@ impl ZRpcOps for LightRpcClient {
                 hash: Vec::new(),
             }),
         });
-        let mut response = self
-            .get_live_client()
-            .await?
+        let mut response = selfi
             .get_block_range(request)
             .await
             .map_to_mm(UpdateBlocksCacheErr::GrpcError)?
             .into_inner();
         // without Pin method get_mut is not found in current scope
         while let Some(block) = Pin::new(&mut response).get_mut().message().await? {
             debug!("Got block {:?}", block);
-            on_block(block)?;
+            let height = u32::try_from(block.height)
+                .map_err(|_| UpdateBlocksCacheErr::InternalError("Block height too large".to_string()))?;
```

Comment:
`block.height` is retrieved from Grpc server, so if it's larger than `u32::MAX`, it's because the server sent a larger number and the error is not due to internal problem, in this case the error should probably be `DecodeError`.

Reply:
thanks. done